src/webgpu/api/operation/command_buffer/programmable/state_tracking.spec.ts

```diff
@@ -10,6 +10,266 @@ TODO: for each programmable pass encoder {compute pass, render pass, render bund
 `;
 
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
-import { GPUTest } from '../../../../gpu_test.js';
+import { GPUConst } from '../../../../constants.js';
+import { ProgrammableStateTest } from './programmable_state_test.js';
 
-export const g = makeTestGroup(GPUTest);
+export const g = makeTestGroup(ProgrammableStateTest);
+
+const kEncoderTypes = ['compute pass', 'render pass', 'render bundle'] as const;
```

Comment:
nit: I think you can import `kProgrammableEncoderTypes` instead of redefining this list